src/test/regress/sql/rpt.sql

```diff
@@ -562,6 +562,21 @@ explain (costs off, verbose) insert into t2 (a, b) select i, random() from t;
 explain (costs off, verbose) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0);
 -- ensure we do not break broadcast motion
 explain (costs off, verbose) select * from t1 where 1 <= ALL (select i from t group by i having random() > 0);
+-- ensure we make motion when volatile function in target list of CTE
+explain (costs off, verbose) with cte as (select a, a * random() from generate_series(1, 5) a)
```

Comment:
Where's the CTE? Maybe you should have turned on the GUC and select twice from cte?

Reply:
why should we test Shared Scan if the problem is reproduced in the case of inline?

Reply:
not inlined case is affected too, may be add two tests?